test/jdk/java/awt/font/TestGraphicPlacement.java

```diff
@@ -0,0 +1,199 @@
+                .build()
+                .awaitAndCheck();
+    }
+
+    public TestGraphicPlacement() {
+        setBackground(Color.WHITE);
+
+        strings = new AttributedString[]{
+                makeString(GraphicAttribute.TOP_ALIGNMENT, SHAPE),
+                makeString(GraphicAttribute.BOTTOM_ALIGNMENT, SHAPE),
+                makeString(GraphicAttribute.ROMAN_BASELINE, SHAPE),
+                makeString(GraphicAttribute.TOP_ALIGNMENT, IMAGE),
+                makeString(GraphicAttribute.BOTTOM_ALIGNMENT, IMAGE),
+                makeString(GraphicAttribute.ROMAN_BASELINE, IMAGE),
+        };
+    }
+
+    @Override
+    public Dimension getPreferredSize() {
+        return new Dimension(350, 450);
+    }
+
+    private Image makeImage(int size) {
+        Image img = new BufferedImage(size, size, BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2d = (Graphics2D) img.getGraphics();
+        g2d.setColor(Color.BLUE);
+        g2d.fillRect(0, 0, size, size);
+        g2d.setColor(Color.RED);
+        g2d.drawRect(0, 0, size - 1, size - 1);
+        return img;
+    }
+
+    /**
+     * Create an AttributedString with a descriptive word (TOP, BOTTOM,
+     * or BASELINE) and several graphics of varying sizes following the
+     * word.
+     *
+     * @param alignment the alignment of the graphics
+     * @param image     whether to create Shape or Image graphics
+     */
+    private AttributedString makeString(int alignment, boolean image) {
+        String name;
+        if (alignment == GraphicAttribute.TOP_ALIGNMENT) {
+            name = "TOP";
+        } else if (alignment == GraphicAttribute.BOTTOM_ALIGNMENT) {
+            name = "BOTTOM";
+        } else {
+            name = "BASELINE";
+        }
+
+        // Append the Unicode graphic replacement character to the name.
+        String nameWithUnicode = name.concat("\uFFFC".repeat(GRAPHIC_COUNT));
+
+        AttributedString as = new AttributedString(nameWithUnicode);
+
+        // Make the descriptive text large.
+        as.addAttribute(TextAttribute.SIZE, 48f, 0, name.length());
+
+        // Add the graphic attributes to the end of the AttributedString.
+        for (int i = 0; i < GRAPHIC_COUNT; i++) {
+
+            float size = (i + 1) * BASE_SIZE;
+            GraphicAttribute attribute;
+
+            if (image == IMAGE) {
+                Image img = makeImage((int) size);
+                attribute = new ImageGraphicAttribute(img, alignment);
+            } else {
+                Shape shape = new Rectangle2D.Float(0, 0, size, size);
+                attribute = new ShapeGraphicAttribute(shape,
+                                                      alignment,
+                                                      ShapeGraphicAttribute.STROKE);
+            }
+
+            as.addAttribute(TextAttribute.CHAR_REPLACEMENT,
+                    attribute,
+                    i + name.length(),
+                    i + name.length() + 1);
```

Comment:
```suggestion
            as.addAttribute(TextAttribute.CHAR_REPLACEMENT,
                            attribute,
                            i + name.length(),
                            i + name.length() + 1);
```
Align the wrapped parameters to match the style in lines 158–160 above?